z_infra_provisioning/cloud_infra_center/ocp_upi/roles/configure-installer-and-image/tasks/main.yaml

```diff
@@ -45,13 +45,13 @@
 - name: Remove openshift RHCOS images
   file:
     state: absent
-    path: rhcos-{{ openshift_version }}.{{ openshift_minor_version }}-s390x.raw.gz
+    path: rhcos-{{ openshift_version }}.{{ openshift_minor_version }}-{{ ansible_architecture }}.raw.gz
```

Comment:
why we use ansible_architecture to replace s390x? It makes the playbook complicated since there is only s390x arch with ICIC.

Reply:
This change was from Jay's comment, it's good advice to read the environment variables.

issue 2: after modifying the configure-installer-client playbook to not set ownership to root , after downloading oc, kubectl, and openshift-install … I notice that the architecture /s390x/ is hard coded in the path. It would be better to use the ansible_architecture variable to pull the binary that will run on this architecture.
```
url: [https://mirror.openshift.com/pub/openshift-v4/{{](https://mirror.openshift.com/pub/openshift-v4/%7B%7B) ansible_architecture }}/clients/ocp/{{ openshift_version }}.{{ openshift_minor_version }}/openshift-install-linux.tar.gz
```

Reply:
I'm ok If ansible_architecture is generated by ansible playbook not user assigned. In other words, if there is no addition work from user for this value, that is ok. If the value need user to assign, there is no necessary to set it because ICIC works on s390x arch only.

Reply:
This environment was reading from user's Ansible host, no needs user to set.